development/openstack-base-focal-wallaby/README.md

```diff
@@ -1,269 +1,371 @@
+and all external networks (North/South traffic).
 
-Servers should have:
+> **Important**: The three MAAS nodes are needed for the actual OpenStack
+  cloud; they do not include the Juju controller. You actually need a minimum
+  of four nodes. The controller node however can be a smaller system (1 CPU and
+  4 GiB memory). Juju [constraints][juju-constraints-controller] can be used to
+  target this smaller system at controller-creation time.
 
- - A minimum of 8GB of physical RAM.
- - Enough CPU cores to support your capacity requirements.
- - Two disks (identified by /dev/sda and /dev/sdb); the first is used by MAAS for the OS install, the second for Ceph storage.
- - Two cabled network ports on eno1 and eno2 (see below).
+## Topology
 
-Servers should have two physical network ports cabled; the first is used for general communication between services in the Cloud, the second is used for 'public' network traffic to and from instances (North/South traffic) running within the Cloud.
+* 3 MAAS nodes, with each hosting one of the following:
+    * Ceph storage
+    * Nova Compute
+    * NTP
 
-## Components
+* LXD containers for the following (distributed among the 3 MAAS nodes):
+    * Ceph monitors (x3)
+    * Ceph RADOS Gateway
+    * Cinder
+    * Glance
+    * Horizon
+    * Keystone
+    * MySQL8 (x3)
+    * Neutron
+    * Nova Cloud Controller
+    * OVN (x3)
+    * Placement
+    * RabbitMQ
+    * Vault
 
- - 3 Nodes for Nova Compute and Ceph, with RabbitMQ, MySQL, Keystone, Glance, Neutron, OVN, Nova Cloud Controller, Ceph RADOS Gateway, Cinder and Horizon under LXC containers.
+## Download the bundle
 
-All physical servers (not LXC containers) will also have NTP installed and configured to keep time in sync.
+If not already done, clone the [openstack-bundles][openstack-bundles]
+repository:
 
-## Deployment
+    git clone https://github.com/openstack-charmers/openstack-bundles ~/openstack-bundles
 
-With a Juju controller bootstrapped on a MAAS cloud with no network spaces
-defined, a basic non-HA cloud can be deployed with the following command:
+The stable and development bundles are found under the `stable/openstack-base`
+and `development` directories respectively.
 
-    juju deploy bundle.yaml
+Overlay bundles are available under `stable/overlays`. See the Juju
+documentation on [overlay bundles][juju-overlays].
 
-When network spaces exist in the MAAS cluster, it is necessary to clarify
-and define the network space(s) to which the charm applications will deploy.
-This can be done with an overlay bundle.  An example overlay yaml file is
-provided, which most likely needs to be edited (before deployment) to
-represent the intended network spaces in the existing MAAS cluster.  Example
-usage:
+## Modify the bundle
 
-    juju deploy bundle.yaml --overlay openstack-base-spaces-overlay.yaml
+If using the stable openstack-base bundle, the file to modify is
+`~/openstack-bundles/stable/openstack-base/bundle.yaml`.
```

Comment:
Make this a relative path: ./ rather than ~/

./openstack-bundles/stable/openstack-base/bundle.yaml